src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelThread.cs

```diff
@@ -226,11 +206,17 @@ public void Send(Action<ListenerSecondary> callback, ListenerSecondary state)
 
         private void PostCloseHandle(Action<IntPtr> callback, IntPtr handle)
         {
-            lock (_workSync)
+            _closeHandleQueue.Enqueue(new CloseHandle { Callback = callback, Handle = handle });
+            WakeUpLoop();
+        }
+
+        private void WakeUpLoop()
+        {
+            if (_loopIdle)
             {
-                _closeHandleAdding.Enqueue(new CloseHandle { Callback = callback, Handle = handle });
+                _loopIdle = false;
+                _post.Send();
```

Comment:
Thought about this for a while; only performant way I can think of is to hit it with a sledgehammer.